src/vcSceneLayer.cpp

```diff
@@ -119,19 +119,17 @@ void vcSceneLayer_LoadNodeJob(void *pData)
     return;
 
   vcSceneLayer_LoadNodeJobData bestJob = {};
-  udLockMutex(gSceneLayer.loadQueue.pLock);
 
-  gSceneLayer.loadQueue.queue.PopFront(&bestJob);
-
-  // Main thread main may have destroyed this scene layer
-  if (!bestJob.validMemory)
   {
-    udReleaseMutex(gSceneLayer.loadQueue.pLock);
-    return;
-  }
+    udScopeLock scopeLock(gSceneLayer.loadQueue.pLock);
```

Comment:
Haven't seen any other references to this particular lock

Reply:
Thought I'd try it out. Made this block a bit nicer.